go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -32,6 +32,14 @@ type PluginEntry struct {
 	// support the same task type. This must be a subset of RegisteredTaskTypes and at most one default per task type
 	// is supported.
 	DefaultForTaskTypes []pluginsCore.TaskType
+	// Returns a new KubeClient to be used instead of the internal controller-runtime client.
+	NewKubeClient func(ctx context.Context) (pluginsCore.KubeClient, error)
```

Comment:
Should we call this - `customKubeClient`

Reply:
renamed